common/src/main/java/io/netty/util/ResourceLeakDetectorFactory.java

```diff
@@ -57,30 +57,44 @@ public static void setResourceLeakDetectorFactory(ResourceLeakDetectorFactory fa
     /**
      * Returns a new instance of a {@link ResourceLeakDetector} with the given resource class.
      *
-     * @param resource - the resource class used to initialize the {@link ResourceLeakDetector}
-     * @param <T> - the type of the resource class
-     * @return - a new instance of {@link ResourceLeakDetector}
+     * @param resource the resource class used to initialize the {@link ResourceLeakDetector}
+     * @param <T> the type of the resource class
+     * @return a new instance of {@link ResourceLeakDetector}
      */
     public final <T> ResourceLeakDetector<T> newResourceLeakDetector(Class<T> resource) {
-        return newResourceLeakDetector(resource, ResourceLeakDetector.DEFAULT_SAMPLING_INTERVAL, Long.MAX_VALUE);
+        return newResourceLeakDetector(resource, ResourceLeakDetector.DEFAULT_SAMPLING_INTERVAL);
     }
 
     /**
+     * @deprecated Use {@link #newResourceLeakDetector(Class, int)} instead.
+     * <p>
      * Returns a new instance of a {@link ResourceLeakDetector} with the given resource class.
      *
-     * @param resource - the resource class used to initialize the {@link ResourceLeakDetector}
-     * @param samplingInterval - the interval on which sampling takes place
-     * @param maxActive - the maximum active instances
-     * @param <T> - the type of the resource class
-     * @return - a new instance of {@link ResourceLeakDetector}
+     * @param resource the resource class used to initialize the {@link ResourceLeakDetector}
+     * @param samplingInterval the interval on which sampling takes place
+     * @param maxActive This is deprecated and will be ignored.
+     * @param <T> the type of the resource class
+     * @return a new instance of {@link ResourceLeakDetector}
      */
+    @Deprecated
     public abstract <T> ResourceLeakDetector<T> newResourceLeakDetector(
             Class<T> resource, int samplingInterval, long maxActive);
 
     /**
+     * Returns a new instance of a {@link ResourceLeakDetector} with the given resource class.
+     *
+     * @param resource the resource class used to initialize the {@link ResourceLeakDetector}
+     * @param samplingInterval the interval on which sampling takes place
+     * @param <T> the type of the resource class
+     * @return a new instance of {@link ResourceLeakDetector}
+     */
+    public abstract <T> ResourceLeakDetector<T> newResourceLeakDetector(Class<T> resource, int samplingInterval);
```

Comment:
@Scottmitch this is a breaking change. We can not add a new abstract method as the class is public.